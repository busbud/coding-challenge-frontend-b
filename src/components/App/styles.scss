
@import '../../shared-styles/variables';

.header {
  text-align: center;
}

.nav {
  width: $content-width;
  margin: 0 auto;
  padding: $padding;
  text-align: center;
}

.content {
  width: $content-width;
  margin: $padding auto;
}

.navItem {
  display: inline-block;
}

.link {
  padding: $padding;
  color: #00a68f;
  font-size: 18px;
}

.logo {
  display: block;
  width: 252px;
  height: 252px;
  margin: $padding auto;
  background: url('../../images/icon-kytLogo_large-252x252.svg');
}
