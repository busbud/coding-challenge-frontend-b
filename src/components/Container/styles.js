import styled from 'styled-components'

export const StyledContainer = styled.div`
  padding: 0 10px;
  margin: 0 auto;
  max-width: 620px;
  width: 100%;
`
