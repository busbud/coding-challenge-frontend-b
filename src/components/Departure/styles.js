import styled from 'styled-components'

export const StyledCity = styled.span`
  color: ${props => props.primary ? '#1f242b' : 'rgba(32, 37, 43, .35)'};
  margin-left: .57rem;
`

export const StyledCityName = styled.span`
  font-size: 16px;
  font-weight: 500;

  &::after {
    content: " - ";
  }
`

export const StyledDeparture = styled.div`
  background-color: #ffffff;
  border-radius: .3rem;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, .24);
  display: flex;
  flex-direction: column;
  margin: 1.42rem 0;
  position: relative;
`

export const StyledDepartureContent = styled.div`
  flex: 1 1 auto;
  padding: 1.14rem;
`

export const StyledDepartureSchedule = styled.div`
  font-size: 14px;
  line-height: 21px;
  margin-bottom: 4px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

export const StyledHeader = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: .57rem;
`

export const StyledOperator = styled.div`
  > img {
    height: 25px;
    width: auto;
  }
`

export const StyledPrice = styled.div`
  color: #f19020;
  font-size: 1.71rem;
  font-weight: 700;
  line-height: 1.71rem;
  text-align: right;
`

export const StyledTime = styled.strong`
  color: ${props => props.primary ? '#127ccb' : 'rgba(32, 37, 43, .35)'};
  display: inline-block;
  font-size: 16px;
  font-weight: 700;
  width: 5em;
`