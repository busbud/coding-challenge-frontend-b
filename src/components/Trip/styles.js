export default theme => ({
  departure: {
    textAlign: 'left',
  },
  duration: {
    textAlign: 'center',
  },
  arrival: {
    textAlign: 'right',
  },
  actions: {
    display: 'flex',
  },
  expand: {
    transform: 'rotate(0deg)',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
    marginLeft: 'auto',
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
});
