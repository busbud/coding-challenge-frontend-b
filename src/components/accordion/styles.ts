// Packages
import styled, { css } from 'styled-components'

import * as L from 'layout'

export const ContentWrapper = styled.div<{ maxHeight: string | undefined }>`
  ${({ theme, maxHeight }) => css`
    background-color: ${theme.colors.white};
    overflow: hidden;
    transition: ${theme.transition.ease};
    max-height: ${maxHeight || 0};

    & > * {
      transition: ${theme.transition.ease};
      opacity: ${maxHeight ? 1 : 0};
    }
  `}
`

export const TriggerWrapper = styled(L.Cluster)`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    transition: ${theme.transition.ease};
  `}
`

export const Wrapper = styled(L.Box)<{ withBorder?: boolean }>`
  ${({ theme, withBorder }) => css`
    border: ${withBorder && `1px solid ${theme.colors.muted}`};
    padding: 1rem;
    cursor: pointer;
    box-sizing: border-box;
    box-shadow: ${theme.shadow.small};
  `}
`
