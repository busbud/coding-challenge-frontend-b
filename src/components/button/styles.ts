import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { darken } from 'polished'

type Props = {
  rounded?: boolean
  circle?: boolean
  primary?: boolean
  secondary?: boolean
  skyBlue?: boolean
  disabled?: boolean
}

const modifiers = {
  rounded: (theme) => css`
    border-radius: ${theme.border.rounded};
    ${media.greaterThan('medium')`
      padding: 1.3rem ${theme.spaces.large};
    `}
  `,
  circle: (theme) => css`
    border-radius: ${theme.border.circle};
    width: 32px;
    height: 32px;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;
  `,
  primary: (theme) => css`
    background-color: ${theme.colors.primary};
    &:hover {
      background: ${darken(0.1, theme.colors.primary)};
    }
  `,
  secondary: (theme) => css`
    background-color: ${theme.colors.secondary};
    &:hover {
      background: ${darken(0.1, theme.colors.secondary)};
    }
  `,
  skyBlue: (theme) => css`
    background-color: ${theme.colors.blue};
    &:hover {
      background: ${darken(0.1, theme.colors.blue)};
    }
  `,
  disabled: (theme) => css`
    background-color: ${theme.colors.muted};
    cursor: not-allowed;
    &:hover {
      background: ${darken(0.1, theme.colors.muted)};
    }
  `
}

export const ButtonWrapper = styled.button<Props>`
  ${({ theme, rounded, circle, primary, secondary, skyBlue, disabled }) => css`
    background-color: ${theme.colors.gray};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    padding: 1rem ${theme.spaces.small};
    font-size: ${theme.font.sizes.base};
    text-align: center;
    white-space: nowrap;
    cursor: pointer;
    transition: ${theme.transition.ease};

    ${rounded && modifiers.rounded(theme)};
    ${circle && modifiers.circle(theme)};
    ${primary && modifiers.primary(theme)};
    ${secondary && modifiers.secondary(theme)};
    ${skyBlue && modifiers.skyBlue(theme)};
    ${disabled && modifiers.disabled(theme)};
  `}
`
