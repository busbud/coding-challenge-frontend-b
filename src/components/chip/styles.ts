import styled, { css } from 'styled-components'

export const Card = styled.article`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: ${theme.spaces.medium};
    color: ${theme.colors.texts};
    font-size: ${theme.font.sizes.xsmall};
    border-radius: ${theme.border.radius};
  `}
`

export const TitleWrapper = styled.div`
  ${({ theme }) => css`
    border-left: 6px solid ${theme.colors.primary};
    padding-left: ${theme.spaces.xsmall};
    margin-bottom: ${theme.spaces.xsmall};
  `}
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: bold;
  `}
`

export const SubTitle = styled.h4`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    font-weight: normal;
    color: ${theme.colors.gray};
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    p + p {
      margin-top: ${theme.spaces.small};
    }
  `}
`
