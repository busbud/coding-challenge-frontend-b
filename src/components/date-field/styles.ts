// Packages
import styled, { css } from 'styled-components'
import { rgba } from 'polished'

export const Overlay = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    background: ${theme.colors.white};
    box-shadow: ${rgba(theme.colors.muted, 1)};
    border: 2px solid ${rgba(theme.colors.gray, 0.3)};
    border-radius: ${theme.border.radius};

    position: absolute;
    transform: translateY(4px);
    z-index: 100;
  `}
`

export const OverlayToolbar = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    display: flex;
    flex-direction: row;
    padding-left: ${theme.spaces.xsmall};
    padding-right: ${theme.spaces.xsmall};
    padding-top: ${theme.spaces.xsmall};
  `}
`

export const LibStyles = styled.div`
  ${({ theme }) => css`
    .DayPicker {
      display: inline-block;
      font-size: ${theme.font.sizes.small};
      margin-top: 0.4rem;
    }

    .DayPicker-wrapper {
      position: relative;
      flex-direction: row;
      padding-bottom: 1rem;

      -webkit-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;

      user-select: none;
    }

    .DayPicker-Months {
      display: flex;
      flex-wrap: wrap;
      justify-content: center;
    }

    .DayPicker-Month {
      display: table;
      margin: 0 1rem;
      margin-top: 1rem;
      border-spacing: 0;
      border-collapse: collapse;

      -webkit-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;

      user-select: none;
    }

    .DayPicker-NavButton {
      position: absolute;
      top: 1rem;
      right: 1.5rem;
      left: auto;

      display: inline-block;
      margin-top: 2px;
      width: 1.25rem;
      height: 1.25rem;
      background-position: center;
      background-size: 50%;
      background-repeat: no-repeat;
      color: ${theme.colors.primary};
      cursor: pointer;
    }

    .DayPicker-NavButton:hover {
      opacity: 0.8;
    }

    .DayPicker-NavButton--prev {
      margin-right: 1.5rem;
      border-right: 2px solid ${rgba(theme.colors.gray, 0.3)};
      border-bottom: 2px solid ${rgba(theme.colors.gray, 0.3)};
      height: 12px;
      width: 12px;
      transform: rotate(130deg);
    }

    .DayPicker-NavButton--next {
      border-right: 2px solid ${rgba(theme.colors.gray, 0.3)};
      border-bottom: 2px solid ${rgba(theme.colors.gray, 0.3)};
      height: 12px;
      width: 12px;
      transform: rotate(-45deg);
    }

    .DayPicker-NavButton--interactionDisabled {
      display: none;
    }

    .DayPicker-Caption {
      display: table-caption;
      margin-bottom: 0.5rem;
      padding: 0 0.5rem;
      text-align: left;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Caption > div {
      font-weight: 500;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Weekdays {
      display: table-header-group;
      margin-top: 1rem;
    }

    .DayPicker-WeekdaysRow {
      display: table-row;
    }

    .DayPicker-Weekday {
      display: table-cell;
      padding: 0.5rem;
      color: ${theme.colors.gray};
      text-align: center;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Weekday abbr[title] {
      border-bottom: none;
      text-decoration: none;
    }

    .DayPicker-Body {
      display: table-row-group;
    }

    .DayPicker-Week {
      display: table-row;
    }

    .DayPicker-Day {
      display: table-cell;
      padding: 0.5rem;
      border-radius: 50%;
      vertical-align: middle;
      text-align: center;
      cursor: pointer;
    }

    .DayPicker-WeekNumber {
      display: table-cell;
      padding: 0.5rem;
      min-width: 1rem;
      border-right: 1px solid ${theme.colors.muted};
      color: ${theme.colors.gray};
      vertical-align: middle;
      text-align: right;
      font-size: ${theme.font.sizes.small};
      cursor: pointer;
    }

    .DayPicker--interactionDisabled .DayPicker-Day {
      cursor: default;
    }

    .DayPicker-Footer {
      padding-top: 0.5rem;
    }

    .DayPicker-TodayButton {
      border: none;
      background-color: transparent;
      background-image: none;
      box-shadow: none;
      color: ${theme.colors.blue};
      font-size: ${theme.font.sizes.small};
      cursor: pointer;
    }

    .DayPicker-Day--today {
      color: ${theme.colors.blue};
      font-weight: 700;
    }

    .DayPicker-Day--outside {
      color: ${theme.colors.gray};
      cursor: default;
    }

    .DayPicker-Day--disabled {
      color: ${rgba(theme.colors.gray, 0.7)};
      cursor: default;
    }

    .DayPicker-Day--sunday {
      background-color: #f7f8f8;
    }

    .DayPicker-Day--sunday:not(.DayPicker-Day--today) {
      color: #dce0e0;
    }

    .DayPicker:not(.DayPicker--interactionDisabled)
      .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
      background-color: ${rgba(theme.colors.blue, 0.7)};
      color: ${theme.colors.white};
    }

    .DayPickerInput {
      display: inline-block;
      width: 100%;
    }

    .DayPickerInput-OverlayWrapper {
      position: relative;
    }

    .Selectable
      .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside) {
      background-color: ${theme.colors.blue};
      color: ${theme.colors.white};
    }

    .Selectable .DayPicker-Day {
      border-radius: 50%;
    }
  `}
`
