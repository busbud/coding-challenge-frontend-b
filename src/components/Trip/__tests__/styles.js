import { createMuiTheme } from 'material-ui/styles';
import styles from '../styles';

const theme = createMuiTheme({});

describe('component | Trip | styles', () => {
  it('should be defined', () => {
    // given
    const css = styles(theme);

    // then
    expect(css).toBeDefined();
  });

  it('should match snapshot', () => {
    // given
    const css = styles(theme);

    // then
    expect(css).toMatchSnapshot();
  });

  it('should have proper rules according to provided theme', () => {
    // given
    const css = styles(theme);
    const transition = css.expand.transition;

    // then
    expect(transition).toBe(
      theme.transitions.create('transform', {
        duration: theme.transitions.duration.shortest,
      }),
    );
  });
});
