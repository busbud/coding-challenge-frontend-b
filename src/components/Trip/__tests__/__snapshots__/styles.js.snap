// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | Trip | styles should match snapshot 1`] = `
Object {
  "actions": Object {
    "display": "flex",
  },
  "arrival": Object {
    "textAlign": "right",
  },
  "departure": Object {
    "textAlign": "left",
  },
  "duration": Object {
    "textAlign": "center",
  },
  "expand": Object {
    "marginLeft": "auto",
    "transform": "rotate(0deg)",
    "transition": "transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
  },
  "expandOpen": Object {
    "transform": "rotate(180deg)",
  },
}
`;
